annotated_test.go

```diff
@@ -65,6 +67,196 @@ func TestAnnotated(t *testing.T) {
 	})
 }
 
+type asStringer struct {
+	name string
+}
+
+func (as *asStringer) String() string {
+	return as.name
+}
+
+func TestAnnotatedAs(t *testing.T) {
+	t.Parallel()
+	type in struct {
+		fx.In
+
+		S fmt.Stringer `name:"goodStringer"`
+	}
+	type myStringer interface {
+		String() string
+	}
+
+	newAsStringer := func() *asStringer {
+		return &asStringer{
+			name: "a good stringer",
+		}
+	}
+
+	tests := []struct {
+		desc    string
+		provide fx.Option
+		invoke  interface{}
+	}{
+		{
+			desc: "provide a good stringer",
+			provide: fx.Provide(
+				fx.Annotate(newAsStringer, fx.As(new(fmt.Stringer))),
+			),
+			invoke: func(s fmt.Stringer) {
+				assert.Equal(t, s.String(), "a good stringer")
+			},
+		},
+		{
+			desc: "provide with multiple types As",
+			provide: fx.Provide(fx.Annotate(func() (*asStringer, *bytes.Buffer) {
+				buf := make([]byte, 1)
+				b := bytes.NewBuffer(buf)
+				return &asStringer{name: "stringer"}, b
+			}, fx.As(new(fmt.Stringer), new(io.Writer)))),
+			invoke: func(s fmt.Stringer, w io.Writer) {
+				w.Write([]byte(s.String()))
+			},
+		},
+		{
+			desc: "provide as with result annotation",
+			provide: fx.Provide(
+				fx.Annotate(func() *asStringer {
+					return &asStringer{name: "stringer"}
+				},
+					fx.ResultTags(`name:"goodStringer"`),
+					fx.As(new(fmt.Stringer))),
```

Comment:
did we run gofmt here? indentation seems off.

Reply:
Idk why but gofmt is the thing that rewrote it this way.